### ISSUE TITLE
Consider State Management solution to refactor `useInkathon` into smaller hooks

### ISSUE THREAD

**Issue by wottpal** [scio-labs/use-inkathon#6]

**Comment by nicolad** [comment]
is this still relevant, since u already use context API from react?

**Comment by wottpal** [comment]
Yep, as i can't easily access this provider context outside hooks.. and it's already huge right now and i see no straightforward way splitting it up w/o using some other state management solution.

**Comment by nicolad** [comment]
Have you used smth like recoil?

https://recoiljs.org/

I am using `react-query` and `react-hook-form` in prod, so I don't really need dedicated state management.

**Comment by jalooc** [comment]
Hey, sleek lib @wottpal, well done 👍 Looking forward to trying it out.

In terms of React state management solutions in which you want an ability to access state outside of hooks, or even React components, I can recommend https://github.com/pmndrs/zustand - really simple to use and similar to Redux in principles, but without its boilerplate.

**Comment by wottpal** [comment]
Thanks so much @jalooc. Would really love to hear more about your experience with the lib or in general what you are up to in the ink! space :) (We also have a tg group https://t.me/inkathon).

And I know Zustand and it's awesome! Recently I've also started using https://jotai.org/ which is dope as well, even though it follows a different paradigm than Zustand/Redux.

**Comment by niklasp** [comment]
i wouldn't introduce a state management library and leave that to the users. let's rely solely on react context for state management. in my former apps i was separating my providers into two internally:

1. extension provider (browser extension related stuff: accounts, activeAccount, signer, ...)
2. chain provider (polkadot api related stuff: api, activeChain, switchActiveChain)
